rules/app.bzl

```diff
@@ -12,6 +13,36 @@ _IOS_APPLICATION_KWARGS = [
     "launch_storyboard",
 ]
 
+def write_info_plists_if_needed(name, plists):
+    """
+    Writes info plists for an app if needed.
+
+    Given a list of infoplists, will write out any plists that are passed as a
+    dict, and will add a default app Info.plist if no non-dict plists are passed.
+
+    Args:
+        name: The name of the app target these infoplists are for.
+        plists: A list of either labels or dicts.
```

Comment:
Can you explain what `labels` would look like and when it would be used? It would be the path to an infoplist file? It looks like you are exercising a single dict path in the test that was added. Worth adding a test that includes the other form? Or does that already have coverage?

Reply:
The label would either be a path to a source plist file, or the label of a generated file. Bazel in general doesn't understand paths, instead using labels to refer to files when instantiating rules, and every source file implicitly gets a label that matches its path